chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/derivation/transformer/rules/TransformSealedHierarchyToSealedHierarchyRuleModule.scala

```diff
@@ -84,14 +84,57 @@ private[compiletime] trait TransformSealedHierarchyToSealedHierarchyRuleModule {
+                            ) >>
+                              deriveRecursiveTransformationExpr[fromSubtype.Underlying, To](fromSubtypeExpr)
+                          // We're constructing:
+                          // case fromSubtypeExpr: $fromSubtype => ${ derivedToSubtype } // or ${ derivedResultToSubtype } // using fromSubtypeExpr.value
+                          lazy val fromSubtypeUnwrappedToSubtype =
+                            fromSubtype.Underlying match {
+                              case WrapperClassType(fromSubtypeInner) =>
+                                import fromSubtypeInner.{Underlying as FromSubtypeInner, value as wrapper}
+                                Some(
+                                  DerivationResult.log(
+                                    s"Falling back on ${Type.prettyPrint[fromSubtypeInner.Underlying]} to ${Type
+                                        .prettyPrint[toSubtype.Underlying]} (source subtype unwrapped)"
+                                  ) >>
+                                    deriveRecursiveTransformationExpr[
+                                      fromSubtypeInner.Underlying,
+                                      toSubtype.Underlying
+                                    ](
+                                      wrapper.unwrap(fromSubtypeExpr)
+                                    ).map(_.map(toUpcast))
+                                )
+                              case _ => None
+                            }
+                          // We're constructing:
+                          // case fromSubtypeExpr: $fromSubtype => Subtype(${ derivedToSubtypeInner } // or ${ derivedResultToSubtypeInner }.map(Subtype)
+                          lazy val fromSubtypeToSubtypeUpwrapped = toSubtype.Underlying match {
+                            case WrapperClassType(toSubtypeInner) =>
+                              import toSubtypeInner.{Underlying as FromSubtypeInner, value as wrapper}
+                              Some(
+                                DerivationResult.log(
+                                  s"Falling back on ${Type.prettyPrint[fromSubtype.Underlying]} to ${Type.prettyPrint[toSubtypeInner.Underlying]} (target subtype unwrapped)"
+                                ) >>
+                                  deriveRecursiveTransformationExpr[fromSubtype.Underlying, toSubtypeInner.Underlying](
+                                    fromSubtypeExpr
+                                  ).map(_.map(wrapper.wrap)).map(_.map(toUpcast))
                               )
-                            )
+                            case _ => None
+                          }
+
+                          subtypeToSubtype
+                            .orElse(subtypeToTarget)
+                            .orElseOpt(fromSubtypeUnwrappedToSubtype)
+                            .orElseOpt(fromSubtypeToSubtypeUpwrapped)
```

Comment:
How about `fromSubtypeUnwrappedToBaseType`? Looks like it would complete all the cases, but we would first need a motivating test.

Reply:
For now I changed it to
```
                          fromSubtypeIntoToSubtype
                            .orElse(fromSubtypeIntoToType)
                            .orElseOpt(fromSubtypeUnwrappedIntoToSubtype)
                            .orElseOpt(fromSubtypeIntoToSubtypeUnwrapped)
```